api/src/main/java/org/apache/iceberg/view/ViewVersion.java

```diff
@@ -74,6 +76,14 @@ default String operation() {
   /** The query output schema at version create time, without aliases */
   int schemaId();
 
+  /** The default catalog when the view is created. */
+  @Nullable
+  String defaultCatalog();
+
+  /** The default namespace to use when the SQL does not contain a namespace. */
+  @Nullable
```

Comment:
Is this optional? What if you don't set it? Does resolution fail at runtime for any unqualified table names?

Reply:
Yes this is optional and query resolution would fail at runtime if this isn't set for unqualified names. Engines like Spark or Trino would have to make sure to set this appropriately when the query of the view isn't qualified

Reply:
@nastra, how do you know whether a name is unqualified at runtime?

Reply:
I've looked at how Trino handles this. 
Trino always sets the **catalog** and the **schema name** on view creation (even though both fields are optional). During View creation the underlying SQL is parsed and checked. If the SQL is unqualified, it is checked that the table exists within the **catalog/schema** that the view is created under. If the SQL is qualified, it is checked that the qualified table exists.

When a SELECT against a View is performed, Trino parses the [underlying SQL](https://github.com/trinodb/trino/blob/10d3d1f4ef3f9d60581cbb2b5c4e60249abc6dc7/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L2448-L2451) and then sets the [Session scope for the view](https://github.com/trinodb/trino/blob/10d3d1f4ef3f9d60581cbb2b5c4e60249abc6dc7/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L4578) using the **catalog/schema** from the stored view definition. 

The [parsed SQL](https://github.com/trinodb/trino/blob/10d3d1f4ef3f9d60581cbb2b5c4e60249abc6dc7/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L2448) knows whether the table is qualified/unqualified. If it's qualified, then it's executed against the qualified table name. If it's unqualified, then it's executed against the **catalog/schema** that was known at view definition time.

Reply:
I think that this should be required in all cases.

Reply:
I've updated this and made it required in https://github.com/apache/iceberg/pull/8156